### PR TITLE
DAOS-5998 test: datamover symlink tests

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag-vm: pr,-hw datamover,-hw

Added test case for dcp symlinks, for variants
with and without dereferencing.

Signed-off-by: Dalton Bohning <daltonx.bohning@intel.com>

### CODE REVIEW

File: src/tests/ftest/datamover/dm_posix_symlinks.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        """JIRA id: DAOS-5998
+        Test Description:
+            Tests copying POSIX symlinks with dcp.
+        :avocado: tags=all,full_regression
+        :avocado: tags=datamover,dcp
+        :avocado: tags=dm_posix_symlinks,dm_posix_symlinks_dcp
+        """
+        self.run_dm_posix_symlinks("DCP")
+
+    def run_dm_posix_symlinks(self, tool):
```
Comment (shimizukko) [inline src/tests/ftest/datamover/dm_posix_symlinks.py:29]: Please add description for tool.

File: src/tests/ftest/datamover/dm_posix_symlinks.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+            # Just compare against the original
+            diff_src = src_posix_path
+
+        # DAOS -> DAOS
+        dst_daos_dir = self.new_daos_test_path(False)
+        self.run_datamover(
+            test_desc + " (DAOS->DAOS)",
+            "DAOS", src_daos_dir, pool, cont,
+            "DAOS", dst_daos_dir, pool, cont)
+        self.run_diff(diff_src, cont.path.value + dst_daos_dir, do_deref)
```
Comment (shimizukko) [inline src/tests/ftest/datamover/dm_posix_symlinks.py:124]: I believe you're calling diff to verify that the copying has been done successfully. How are you checking the output of the diff command?

File: src/tests/ftest/datamover/dm_posix_symlinks.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/python
+'''
+  (C) Copyright 2020-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+'''
+from data_mover_test_base import DataMoverTestBase
+from os.path import join
+
```
Comment (shimizukko) [inline src/tests/ftest/datamover/dm_posix_symlinks.py:9]: I believe we add 2 lines before/between classes.

**Review by shimizukko — APPROVED** [review]
LGTM

File: src/tests/ftest/datamover/dm_posix_symlinks.py (hunk 2 above)
Comment (daltonbohning) [reply src/tests/ftest/datamover/dm_posix_symlinks.py:124]: `diff` returns 0 if there are no differences found, and 1 if there are differences found. In the case that it returns 1, `self.execute_cmd` will call `self.fail`. If needed, we could be more explicit and pass `--report-identical-files` and check the diff output for "Files <src> and <dst> are identical"

**Review by shimizukko — APPROVED** [review]
LGTM

**Review by dsikich — APPROVED** [review]

**Pull request merged**